subsys/settings/Kconfig

```diff
@@ -35,6 +35,7 @@ config SETTINGS_ENCODE_LEN
 
 choice SETTINGS_BACKEND
 	prompt "Storage back-end"
+	default SETTINGS_RAM
```

Comment:
The current default `SETTINGS_NONE` should either be removed or kept as the default. IMO `SETTINGS_NONE` should be kept, making the RAM backend an explicit choice for the user.

Also, as the RAM backend would be mostly used for testing purposes, it makes sense to print out a warning during the build.